internal/datastore/common/chunkbytes.go

```diff
@@ -0,0 +1,379 @@
+}
+
+// ChunkedBytesExecutor defines the interface for creating transactions for chunked byte operations.
+type ChunkedBytesExecutor interface {
+	// BeginTransaction starts a new transaction for chunked byte operations.
+	BeginTransaction(ctx context.Context) (ChunkedBytesTransaction, error)
+
+	// ExecuteRead executes a SELECT query and returns the results as a map of chunk index to chunk data.
+	ExecuteRead(ctx context.Context, builder sq.SelectBuilder) (map[int][]byte, error)
+}
+
+// WriteMode defines how chunked data should be written.
+type WriteMode int
+
+const (
+	// WriteModeInsertWithTombstones inserts new chunks and marks old chunks with a tombstone.
+	// Requires TombstoneColumn to be set in config.
+	WriteModeInsertWithTombstones WriteMode = iota
+
+	// WriteModeDeleteAndInsert deletes all existing chunks for the key before inserting new ones.
+	// Useful for replacing data completely.
+	WriteModeDeleteAndInsert
+)
+
+// SQLByteChunkerConfig contains the configuration for creating a SQLByteChunker.
+type SQLByteChunkerConfig[T any] struct {
+	// TableName is the name of the table storing the chunked data.
+	TableName string
+
+	// NameColumn is the column name that stores the identifier for the byte data.
+	NameColumn string
+
+	// ChunkIndexColumn is the column name that stores the chunk index (0-based).
+	ChunkIndexColumn string
+
+	// ChunkDataColumn is the column name that stores the chunk bytes.
+	ChunkDataColumn string
+
+	// MaxChunkSize is the maximum size in bytes for each chunk.
+	MaxChunkSize int
+
+	// PlaceholderFormat is the placeholder format for SQL queries (e.g., sq.Question, sq.Dollar).
+	PlaceholderFormat sq.PlaceholderFormat
+
+	// Executor is the executor for running SQL queries.
+	Executor ChunkedBytesExecutor
+
+	// WriteMode defines how chunked data should be written (insert-with-tombstones or delete-and-insert).
+	WriteMode WriteMode
+
+	// CreatedAtColumn is the column name that stores when a row was created (alive timestamp/transaction ID).
+	// Required when WriteMode is WriteModeInsertWithTombstones.
+	CreatedAtColumn string
+
+	// DeletedAtColumn is the column name that stores when a row was deleted (tombstone timestamp/transaction ID).
+	// Required when WriteMode is WriteModeInsertWithTombstones.
+	DeletedAtColumn string
+
+	// AliveValue is the value used to indicate a row has not been deleted yet (typically max int).
+	// Required when WriteMode is WriteModeInsertWithTombstones.
+	AliveValue T
+}
+
+// SQLByteChunker provides methods for reading and writing byte data
+// that is chunked across multiple rows in a SQL table.
+type SQLByteChunker[T any] struct {
+	tableName         string
+	nameColumn        string
+	chunkIndexColumn  string
+	chunkDataColumn   string
+	maxChunkSize      int
+	placeholderFormat sq.PlaceholderFormat
+	executor          ChunkedBytesExecutor
+	writeMode         WriteMode
+	createdAtColumn   string
+	deletedAtColumn   string
+	aliveValue        T
```

Comment:
all these fields are already defined in `SQLByteChunkerConfig`. why not just hold a pointer to it? this would reduce duplicate code and prevent you from having to write

```
return &SQLByteChunker[T]{
		tableName:         config.TableName,
		nameColumn:        config.NameColumn,
		chunkIndexColumn:  config.ChunkIndexColumn,
		chunkDataColumn:   config.ChunkDataColumn,
		maxChunkSize:      config.MaxChunkSize,
		placeholderFormat: config.PlaceholderFormat,
		executor:          config.Executor,
		writeMode:         config.WriteMode,
		createdAtColumn:   config.CreatedAtColumn,
		deletedAtColumn:   config.DeletedAtColumn,
		aliveValue:        config.AliveValue,
	}
```

which is dangerous, because if you add a new configuration you have to remember to update that code too